packages/@aws-cdk/aws-cognito/lib/user-pool-client.ts

```diff
@@ -339,6 +347,11 @@ export class UserPoolClient extends Resource implements IUserPoolClient {
   }
 
   public readonly userPoolClientId: string;
+
+  private _generateSecret: boolean | undefined;
```

Comment:
This seems redundant. With a if statement false and undefined will both return as false. But it could also just be written as `private _generateSecret?: boolean;` which is more stylistically consistent.

Reply:
Agreed. Good suggestion. Implemented.